src/CasparCG.ts

```diff
@@ -1,2365 +1,539 @@
-			commandPromiseArray.push(new Promise<IAMCPCommand>((resolve, reject) => {
-				subCommand.resolve = resolve
-				subCommand.reject = reject
-			}))
-		}
-
-		commandPromise = Promise.all(commandPromiseArray)
-		commandPromise.catch((error: any) => {
-			// @todo: global command error handler here
-			this._log(new Error('Command error: ' + error.toString()))
-		})
-
-		switch (priority) {
-			case Priority.NORMAL:
-				this._queuedCommands.push(command)
-				break
-			case Priority.HIGH:
-				this._queuedCommandsHighPriority.push(command)
-				break
-			case Priority.LOW:
-				this._queuedCommandsLowPriority.push(command)
-				break
-		}
-
-		this._log(`New command added, "${command.name}". ${this.commandQueueLength} command(s) in command queues.`)
-		command.status = IAMCPStatus.Queued
-
-		this._executeNextCommand()
-		return new Promise((outerResolve, outerReject) => {
-			commandPromise.then((cmds) => [
-				outerResolve(cmds[cmds.length - 1]) // resolve with last executed command
-			]).catch((err) => {
-				outerReject(err)
-			})
+import { BasicCasparCGAPI, Request } from './api'
+import { Commands } from './commands'
+import {
+	LoadbgParameters,
+	LoadParameters,
+	PlayParameters,
+	PauseParameters,
+	ResumeParameters,
+	StopParameters,
+	ClearParameters,
+	CallParameters,
+	SwapParameters,
+	AddParameters,
+	RemoveParameters,
+	PrintParameters,
+	LogLevelParameters,
+	LogCategoryParameters,
+	SetParameters,
+	LockParameters,
+	DataStoreParameters,
+	DataRetrieveParameters,
+	DataListParameters,
+	DataRemoveParameters,
+	CgAddParameters,
+	CgPlayParameters,
+	CgStopParameters,
+	CgNextParameters,
+	CgRemoveParameters,
+	CgClearParameters,
+	CgUpdateParameters,
+	CgInvokeParameters,
+	CgInfoParameters,
+	MixerKeyerParameters,
+	MixerChromaParameters,
+	MixerBlendParameters,
+	MixerInvertParameters,
+	MixerOpacityParameters,
+	MixerBrightnessParameters,
+	MixerSaturationParameters,
+	MixerContrastParameters,
+	MixerLevelsParameters,
+	MixerFillParameters,
+	MixerClipParameters,
+	MixerAnchorParameters,
+	MixerCropParameters,
+	MixerRotationParameters,
+	MixerPerspectiveParameters,
+	MixerMipmapParameters,
+	MixerVolumeParameters,
+	MixerMastervolumeParameters,
+	MixerStraightAlphaOutputParameters,
+	MixerGridParameters,
+	MixerCommitParameters,
+	MixerClearParameters,
+	ChannelGridParameters,
+	ThumbnailListParameters,
+	ThumbnailRetrieveParameters,
+	ThumbnailGenerateParameters,
+	ThumbnailGenerateAllParameters,
+	CinfParameters,
+	ClsParameters,
+	FlsParameters,
+	TlsParameters,
+	VersionParameters,
+	InfoParameters,
+	InfoTemplateParameters,
+	InfoConfigParameters,
+	InfoPathsParameters,
+	InfoSystemParameters,
+	InfoServerParameters,
+	InfoQueuesParameters,
+	InfoThreadsParameters,
+	InfoDelayParameters,
+	DiagParameters,
+	GlInfoParameters,
+	GlGcParameters,
+	ByeParameters,
+	KillParameters,
+	RestartParameters,
+} from './parameters'
+
+export class CasparCG extends BasicCasparCGAPI {
```

Comment:
I'm missing a method for closing the CasparCG connection (`.close()`?).  This would be useful for example when closing one connection and opening another.

Reply:
Not needed for your use case but yes, will add.

ps: just change the host for your use case. it will work exactly like a close() + open() :)